### PR TITLE
Check deprecated method in test of Link

### PR DESCRIPTION
**Pull request by unnonouno**
I fixed test of link to check DeprecationWarning.
merge #2932 first
related to #2545

### CODE REVIEW

**Comment by delta2323** [comment]
I think the label to this PR should be "to-be-backported", not "backport".

**Comment by niboshi** [comment]
#2932 has been merged.
Could you rebase to or merge master?

File: tests/chainer_tests/test_link.py [hunk 1]
```diff
@@ -434,13 +442,21 @@ def zerograd(self):
...
+            self.l3.x = chainer.Parameter()
 
-        self.c1 = chainer.Chain(l1=self.l1)
-        self.c1.add_link('l2', self.l2)
-        self.c2 = chainer.Chain(c1=self.c1)
+        with testing.assert_warns(DeprecationWarning):
+            self.c1 = chainer.Chain(l1=self.l1)
+            self.c1.add_link('l2', self.l2)
+        with testing.assert_warns(DeprecationWarning):
+            self.c2 = chainer.Chain(c1=self.c1)
```
Comment (niboshi) [inline tests/chainer_tests/test_link.py]: How about using `init_scope`?

**Comment by unnonouno** [comment]
I updated it

**Comment by niboshi** [comment]
How about `c1.l2`, `c2.c1`, and `c2.l3`?

**Comment by niboshi** [comment]
I think we should avoid using deprecated functions except in tests for those functions itself.

**Comment by unnonouno** [comment]
`c1.l2` is for test for `add_link` https://github.com/chainer/chainer/pull/3155/files#diff-05835e3a5c0abcd4ac2d0fb29936ac76R478

**Comment by niboshi** [comment]
OK.

**Comment by niboshi** [comment]
LGTM!

**Pull request merged**